static/app/views/setupWizard/index.tsx

```diff
@@ -1,79 +1,50 @@
-import {Component} from 'react';
+import {useEffect, useState} from 'react';
 
-import {Client} from 'sentry/api';
+import Button from 'sentry/components/button';
 import LoadingIndicator from 'sentry/components/loadingIndicator';
 import {t} from 'sentry/locale';
-import withApi from 'sentry/utils/withApi';
+import useApi from 'sentry/utils/useApi';
 
 type Props = {
-  api: Client;
   hash?: boolean | string;
 };
 
-type State = {
-  finished: boolean;
-};
-
-class SetupWizard extends Component<Props, State> {
-  static defaultProps = {
-    hash: false,
-  };
-
-  state: State = {
-    finished: false,
-  };
-
-  UNSAFE_componentWillMount() {
-    this.pollFinished();
-  }
-
-  pollFinished() {
-    return new Promise<void>(resolve => {
-      this.props.api.request(`/wizard/${this.props.hash}/`, {
-        method: 'GET',
-        success: () => {
-          setTimeout(() => this.pollFinished(), 1000);
-        },
-        error: () => {
-          resolve();
-          this.setState({finished: true});
-          setTimeout(() => window.close(), 10000);
-        },
```

Comment:
I don't really understand how this used to work 🤔

Reply:
Agree.. this is a little strange... so when the API returns an error..it means that the setup is done?

Reply:
https://github.com/getsentry/sentry/blob/80ecaaa5a31ffa379f0a0898010be20678b76d7a/src/sentry/api/endpoints/setup_wizard.py#L28-L56

The endpoint didn't offer any great clues either :grim